File: app/src/main/java/com/base/httpmvp/mvpbase/IBaseView.java
package com.base.httpmvp.mvpbase;

import io.reactivex.ObservableTransformer;

/**
 * mvp中的view
 */

public interface IBaseView {

    /**
     * 显示操作进度
     */
    public void showProgress();

    /**
     * 关闭进度
     */
    public void closeProgress();
    /**
     * 绑定生命周期
     * @param <T>
     * @return
     */
    <T> ObservableTransformer<T,T> bindLifecycle();
}

File: app/src/main/java/com/base/httpmvp/mvpbase/IBaseView.kt
package com.base.httpmvp.mvpbase

import androidx.lifecycle.LifecycleOwner
import io.reactivex.ObservableTransformer

/**
 * mvp中的view
 */
interface IBaseView {
    /**
     * 显示操作进度
     */
    fun showProgress()

    /**
     * 关闭进度
     */
    fun closeProgress()

    /**
     * 绑定生命周期
     * @return
    </T> */
    fun bindLifecycle(): LifecycleOwner?
}